src/ConfigDiscoveryTrait.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * @link      http://github.com/zfcampus/zf-development-mode for the canonical source repository
+ * @copyright Copyright (c) 2014-2016 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace ZF\DevelopmentMode;
+
+use RuntimeException;
+
+/**
+ * Shared functionality for the Disable/Enable commands.
+ */
+trait ConfigDiscoveryTrait
+{
+    /**
+     * @var null|array
+     */
+    private $applicationConfig;
+
+    /**
+     * @var string
+     */
+    private $applicationConfigPath = 'config/application.config.php';
+
+    /**
+     * @var string Base name for configuration cache.
+     */
+    private $configCacheBase = 'module-config-cache';
+
+    /**
+     * Removes the application configuration cache file, if present.
+     *
+     * @param string $configCacheFile
+     */
+    private function removeConfigCacheFile($configCacheFile)
+    {
+        if (! $configCacheFile || ! file_exists($configCacheFile)) {
+            return;
+        }
+
+        unlink($configCacheFile);
+    }
+
+    /**
+     * Retrieve the config cache file, if any.
+     *
+     * @return false|string
+     */
+    private function getConfigCacheFile()
+    {
+        $configCacheDir = $this->getConfigCacheDir();
+        $configCacheKey = $this->getConfigCacheKey();
+
+        if (empty($configCacheDir)) {
+            return false;
+        }
+
+        $path = sprintf('%s/%s.', $configCacheDir, $this->configCacheBase);
+
+        if (! empty($configCacheKey)) {
+            $path .= $configCacheKey . '.';
+        }
+
+        return $path . 'php';
+    }
+
+    /**
+     * Return the configured configuration cache directory, if any.
+     *
+     * @return null|string
+     */
+    public function getConfigCacheDir()
+    {
+        $config = $this->getApplicationConfig();
+        if (! isset($config['module_listener_options']['cache_dir'])) {
```

Comment:
`empty()` will raise an error if any one of the keys in the hierarchy is missing, while `isset()` does not. `isset()` will also return `false` if the value in that key is empty, which makes it more useful for this case.